recipe/meta.yaml

```diff
@@ -5,13 +5,47 @@ package:
 build:
   number: 0
 
-test:
-  commands:
-    # Verify the scripts are in-place.
-    {% for state in ["activate", "deactivate"] %}
-    - test -f "${PREFIX}/etc/conda/{{ state }}.d/toolchain_{{ state }}.sh"                        # [unix]
-    - if not exist %PREFIX%\\etc\\conda\\{{ state }}.d\\toolchain_{{ state }}.bat exit 1  # [win]
-    {% endfor %}
+outputs:
+  - name: toolchain
+    # for windows, this is a placebo package.  The default conda-build activation stuff is all we use.
+    # we just don't want people to have to put selectors on their toolchain dep.
+    script: install-toolchain.sh      # [unix]
+    test:
+      commands:
+        # Verify the scripts are in-place.
+        {% for state in ["activate", "deactivate"] %}
+        - test -f "${PREFIX}/etc/conda/{{ state }}.d/toolchain_{{ state }}.sh"                        # [unix]
+        {% endfor %}
+        - echo "I'm on windows, and there's nothing to do here."                                      # [win]
+{% if unix %}
+        - source ${PREFIX}/etc/conda/activate.d/toolchain_activate.sh
+        - if [ -z ${CC+x} ]; then echo "CC is unset after activation" && exit 1; else echo "CC is set to '$CC'"; fi
+        - source ${PREFIX}/etc/conda/deactivate.d/toolchain_deactivate.sh
+        - if [ -z ${CC+x} ]; then echo "CC is unset after deactivation "; else echo "CC is set to '$CC' after deactivation" && exit 1; fi
+
+  - name: toolchain_c_{{ target_platform }}
+    requirements:
+      - {{ pin_subpackage('toolchain', exact=True) }}
+    run_exports:                                                     # [linux]
+      strong:                                                        # [linux]
+        - libgcc-ng >=4.9                                            # [linux]
+
+  - name: toolchain_cxx_{{ target_platform }}
+    requirements:
+      - {{ pin_subpackage('toolchain', exact=True) }}
+    run_exports:
+      strong:
+        - libstdcxx-ng >=4.9                                         # [linux]
+        - libcxx >=4.0                                               # [osx]
```

Comment:
So we currently don't ship these things as we have been using system libraries. Guess we could start doing that with Linux and that would probably help some people who have been wanting this functionality. As we use devtoolset-2, all code is compatible with 4.4+. The LLVM version is based off of 3.6. So would assume the `libcxx` version matches, but am having trouble finding conformation. Given there are some symbols removed in 4.0, we might want to be a little more cautious here.

ref: https://abi-laboratory.pro/tracker/compat_report/libcxx/3.9.1/4.0.0/54f71/abi_compat_report.html

Reply:
Maybe @mingwandroid can tell me that I'm being too paranoid here. ;)